packages/orm/src/relations/ReactiveField.ts

```diff
@@ -95,22 +95,29 @@ export class ReactiveFieldImpl<T extends Entity, H extends ReactiveHint<T>, V>
   }
 
   async load(opts?: { forceReload?: boolean }): Promise<V> {
+    // Even without `forceReload=true`, any explicit calls to `.load()` ==> ensure a fresh value,
+    // because `.get` may have cached the stale/previously-calculated value.
+    //
+    // (Currently `em.populate` / `populateDataLoader` has a hack/escape hatch that ignores calling
+    // RF.load(), to avoid pulling in the RF subgraph over using the materialized value, which ideally would be
+    // replaced with us learning to do the maybeDirty approach.)
     if (!this.isLoaded || opts?.forceReload) {
-      // Even without `forceReload=true`, any explicit calls to `.load()` ==> ensure a fresh value,
-      // because `.get` may have cached the stale/previously-calculated value.
-      //
-      // (Currently `em.populate` / `populateDataLoader` has a hack/escape hatch that ignores calling
-      // RF.load(), to avoid pulling in the RF subgraph over using the materialized value, which ideally would be
-      // replaced with us learning to do the maybeDirty approach.)
-      this.#isCached = false;
```

Comment:
This `isCached=false` being here, outside the `populate` was the core bug b/c we would:

* set `isTest.isCached=false`
* start waiting for `isTest` to be loaded
* while that was happening `isTestChain` would say "hey I can go now"
* `isTestChain` would call `isTest`, and `isTest` knew it still wasn't loaded, but would go ahead and return its stale value (which is not incorrect, as long as we remember to `isTestChain` about the new value later, which we do), *but* we'd also set `isTest.isCached=true` as a side-effect
* when `isTest` was finally loaded, `isTest.get` would run to "update isTest" but see "oh hey I'm already `isCached=true` so just stop